src/tracer.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+import * as Constants from './constants';
+import Span from './span';
+
+/**
+ *
+ */
+export default class Tracer {
+
+    /**
+     * [constructor description]
+     * @return {[type]} [description]
+     */
+    constructor() {
+        this._imp = null;
+    }
+
+    /**
+     * Handle to implementation object.
+     */
+    imp() {
+        return this._imp;
+    }
+
+    /**
+     * TODO: Unclear what the OpenTracing specification wants here.  This
+     * implementation is a best-guess approximation that's likely not right.
+     * The spec refers to text and binary representations, but what does
+     * "binary" mean in JavaScript without making assumptions about protocols,
+     * transports, and platforms?
```

Comment:
The problem is that, as far as I know, there's no standard "byte array" in JavaScript as a language. Yes, really :)

* All of JS supports an `Array` of `number` objects - but this is very inefficient
* Node.js supports `Buffer` (largely to compensate for this)
* The browser does *not* support Buffer
* Some (most?) browsers support [Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array), but I don't know how common this is or how well it would adapt to the use case that this binary format is being required for

https://github.com/opentracing/opentracing-javascript/issues/6 might be a good place to discuss this issue.